common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -460,14 +471,115 @@ private static boolean isAndroid0() {
+                                return Integer.parseInt(parts[1].trim());
+                            }
+                        }
+                    } catch (Throwable cause) {
+                        logger.debug(String.format("Failed to retrieve cache line size, using default: %db",
+                                                   DEFAULT_CACHE_LINE_SIZE), cause);
+                        return DEFAULT_CACHE_LINE_SIZE;
+                    } finally {
+                        if (in != null) {
+                            try {
+                                in.close();
+                            } catch (IOException ignore) {
+                                // ignore on close
+                            }
+                        }
+                    }
+                } else if (isOsx0()) {
+                    Process p = null;
+                    BufferedReader in = null;
+                    try {
+                        p = Runtime.getRuntime().exec(new String[] { "sysctl", "hw.cachelinesize" });
+                        in = new BufferedReader(new InputStreamReader(p.getInputStream(), CharsetUtil.US_ASCII));
+                        String line = in.readLine();
+                        in.close();
+
+                        for (;;) {
+                            try {
+                                int exitCode = p.waitFor();
+                                if (exitCode != 0) {
+                                    line = null;
+                                }
+                                break;
+                            } catch (InterruptedException e) {
+                                Thread.currentThread().interrupt();
+                                break;
+                            }
+                        }
+                        if (line != null) {
+                            String[] parts = StringUtil.split(line, ':');
+                            return Integer.parseInt(parts[1].trim());
+                        }
+                    } catch (Exception e) {
+                        // Failed to run the command.
+                        logger.debug(String.format("Failed to retrieve cache line size, using default: %db",
+                                                   DEFAULT_CACHE_LINE_SIZE), e);
+                        return DEFAULT_CACHE_LINE_SIZE;
+                    } finally {
+                        if (in != null) {
+                            try {
+                                in.close();
+                            } catch (IOException e) {
+                                // Ignore
+                            }
+                        }
+                        if (p != null) {
+                            try {
+                                p.destroy();
+                            } catch (Exception e) {
+                                // Android sometimes triggers an ErrnoException.
```

Comment:
Is it worth logging this?  Seems like a bad thing if the process isn't destroyed?